_posts/2023-05-25-Announcing-Data-Prepper-2.3.0.md

```diff
@@ -0,0 +1,147 @@
+
+```
+processor:
+   - add_entries:
+       entries:
+         - metadata_key: "request_len"
+           value_expression: "length(/request)"
+```
+This configuration adds an attribute to event metadata with the attribute key `request_len` with value equal to the length of the request field in the key. The value can be set using the value, format, or `value_expression` option of entries field.
+
+
+## Amazon S3 sink
+
+Data Prepper now supports saving data to Amazon S3 sinks as Newline delimited JSON (ndjson). Amazon S3 is a popular choice for storing large volumes of data reliably and in a cost-effective way.
+
+Ingesting data into Amazon S3 offers many possibilities for your data pipelines, including some of the following:
+
+* Noisy or uninteresting data can be routed to Amazon S3 and not to OpenSearch to reduce load on your OpenSearch cluster. This can help you save on compute and storage costs.
+* Ingesting data to Amazon S3 creates data that you can use for future processing.
+
+
+## Tail sampling
+
+Data Prepper 2.3.0 supports tail sampling to limit the number of events that are sent to a sink similar to the tail sampling support provided by OpenTelemetry. For details on tail sampling in [OpenTelemetry](https://opentelemetry.io) see the blog ["Tail Sampling with OpenTelemetry: Why it’s useful, how to do it, and what to consider"](https://opentelemetry.io/blog/2022/tail-sampling/).
+
+Tail processing in Data Prepper is supported as an action to the `aggregate` processor. The events are stored in the `aggregate` processor beyond the `group_duration` time until no events are received in the last `wait_period` time. 
+
+For example, the following configuration sends all traces with errors to the sink and non-error events are sampled with the user specified `percent` probabilistic sampler:
+
+```
+trace-normal-pipeline:
+  source:
+    otel_trace_source:
+      ssl: false
+  processor:
+    - trace_peer_forwarder:
+  processor:
+    - aggregate:
+        identification_keys: ["traceId"]
+        action:
+          tail_sampler:
+            percent: 20
+            wait_period: "15s"
+            error_condition: "/traceGroupFields/statusCode == 2"
+        group_duration: "30s"
+ sink:
+    - opensearch
+         hosts: ["https://opensearch:9200"]
+         insecure: true
+         username: "admin"
+         password: "admin"
+         index: sampled-traces
+
+```
+
+## Getting started
+
+* To download Data Prepper, see the [OpenSearch downloads](https://opensearch.org/downloads.html) page.
+* For instructions on how to get started with Data Prepper, see [Getting started with Data Prepper](https://opensearch.org/docs/latest/data-prepper/getting-started/).
+* To learn more about the work in progress for Data Prepper 2.3, see the [Data Prepper roadmap](https://github.com/opensearch-project/data-prepper/projects/1).
```

Comment:
Do we mean 2.3 here or the next version?

Reply:
This should be 2.4.